code/modules/urist/modules/shipbattles/torpedo_warhead.dm

```diff
@@ -0,0 +1,194 @@
+				icon_state = "torpedowarhead-open-mod"
+		else if(riggedstate == CIRCUITRY_RIGGED)
+			to_chat(user, "<span class='notice'>You snip the wire attached to the warhead's detonation circuit.</span>")
+			riggedstate = CIRCUITRY_MODIFIED
+			playsound(src.loc, 'sound/items/Wirecutter.ogg', 50, 1)
+			icon_state = "torpedowarhead-open-mod"
+		else if(riggedstate == CIRCUITRY_MODIFIED)
+			to_chat(user, "<span class='notice'>You carefully undo the modifications to the warhead's circuitry.</span>")
+			riggedstate = CIRCUITRY_EXPOSED
+			playsound(src.loc, 'sound/items/Wirecutter.ogg', 50, 1)
+			icon_state = "torpedowarhead-open"
+		else if(riggedstate == CIRCUITRY_EXPOSED)
+			to_chat(user, "<span class='notice'>You begin to carefully modify the circuitry of the warhead.</span>")
+			if(do_after(user,4 SECONDS))
+				to_chat(user, "<span class='notice'>You have modified the torpedo warhead's internal circuitry. It can now be wired up and attached to something.</span>")
+				riggedstate = CIRCUITRY_MODIFIED
+				playsound(src.loc, 'sound/items/Wirecutter.ogg', 50, 1)
+				icon_state = "torpedowarhead-open-mod"
+		return
+	else if(istype(I, /obj/item/device/multitool))
+		if(riggedstate < CIRCUITRY_EXPOSED)
+			wires.Interact(user)
+	else if(istype(I, /obj/item/stack/cable_coil) && riggedstate == CIRCUITRY_MODIFIED)
+		to_chat(user, "<span class='notice'>You rig a wire from the torpedo warhead's detonator circuit. You can now attach something to it to detonate it remotely.</span>")
+		riggedstate = CIRCUITRY_RIGGED
+		playsound(src.loc, 'sound/items/Wirecutter.ogg', 50, 1)
+	else if(istype(I, /obj/item/device/assembly))
+		if(riggedstate == CIRCUITRY_RIGGED)
+			var/obj/item/device/assembly/A = I
+			to_chat(user, "<span class='warning'>You rig the [A] to the torpedo warhead's detonator circuit!</span>")
+			is_rigged = 1
+			if(!user.unEquip(A, src))
+				return
+			A.forceMove(src)
+			attached_device = I
+			A.holder = src
+			log_and_message_admins("[user] has rigged a torpedo IED.")
+			playsound(src.loc, 'sound/items/Wirecutter.ogg', 50, 1)
+			icon_state = "torpedowarhead-open-mod-armed"
+	return
+
+/obj/item/shipweapons/torpedo_warhead/attack_self(mob/user as mob)
+	..()
+	if(riggedstate < CIRCUITRY_EXPOSED)
+		wires.Interact(user)
+
+/obj/item/shipweapons/torpedo_warhead/proc/process_activation() // uh oh, time to boom
+	detonate()
+
+/obj/item/shipweapons/torpedo_warhead/proc/detonate(var/forced = 0)
+	playsound(src.loc, 'sound/machines/buttonbeep.ogg', 25, 0, 10)
+	if(safety && !forced)
+		visible_message("<span class='danger'>[src] beeps stubbornly, refusing to detonate!</span>")
+		playsound(src.loc, 'sound/machines/buzz-sigh.ogg', 25, 0, 10)
+	if(forced)
+		visible_message("<span class='danger'>[src] pings, begining a short countdown!</span>")
+		playsound(src.loc, 'sound/machines/ping.ogg', 25, 0, 10)
+		playsound(src.loc, 'sound/items/countdown.ogg', 25, 0, 10)
+		spawn(4 SECONDS)
```

Comment:
There's now the rare possibility that one could re-engage the safeties _after_ its set to explode. If you're feeling kind you could give the madman not running away a chance at survival.